### PR TITLE
Adding MLAgility models

### PR DESCRIPTION
**Pull request by danielholanda**

### CODE REVIEW

**Comment by ramkrishna2910** [comment]
Do we want to add GNNs here? Or wait for the fix to the env?

**Comment by jeremyfowers** [comment]
Good opportunity to revisit the directory structure/names?

-`huggingface` is really the pytorch side of huggingface transformers library (rename?)
-`huggingface_tf` is really the Keras side of huggingface transformers library (rename?)
-`torch_hub_vision` is vision models from Torch Hub, but we also have non-vision models from Torch Hub in `torch_hub` (converge?)
-`stable_diffusion` is some stable diffusion sub-models from huggingface diffusers library (rename to `diffusers`? we should probably get more models from `diffusers` as well!)

File: src/mlagility/parser.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    def add_arg(self, parser_obj):
+        parser_obj.add_argument(
+            "--" + self.name, default=self.default, type=self.type, nargs=self.nargs
+        )
+
+
+def parse(valid_args: List):
+
+    # List valid ml-agility args
```
[the remark this answers is not in the document]
Comment (danielholanda) [reply src/mlagility/parser.py:25]: Changed!

[the remark this answers is not in the document]
Comment (danielholanda) [reply src/mlagility/parser.py:25]: Added!

**Comment by danielholanda** [comment]
> Good opportunity to revisit the directory structure/names?
> 
> -`huggingface` is really the pytorch side of huggingface transformers library (rename?) -`huggingface_tf` is really the Keras side of huggingface transformers library (rename?) -`torch_hub_vision` is vision models from Torch Hub, but we also have non-vision models from Torch Hub in `torch_hub` (converge?) -`stable_diffusion` is some stable diffusion sub-models from huggingface diffusers library (rename to `diffusers`? we should probably get more models from `diffusers` as well!)
> 
> edit: I also want to bring in the "popular on huggingface" models - where do those fit into this dir structure?

@jeremyfowers I addressed all of the suggestions regarding renaming/moving files. 

What do you think we should do about adding the popular models? A more updated list that what we have would be nice. How many models should we add here? Also, is that within the scope of this MR?

**Comment by danielholanda** [comment]
> Do we want to add GNNs here? Or wait for the fix to the env?

@ramkrishna2910 I'm ok leaving those here now while we fix the env.

**Comment by jeremyfowers** [comment]
> > Do we want to add GNNs here? Or wait for the fix to the env?
> 
> @ramkrishna2910 I'm ok leaving those here now while we fix the env.

Put the GNNs in `gnn/skip`?

**Comment by jeremyfowers** [comment]
> > Good opportunity to revisit the directory structure/names?
> > -`huggingface` is really the pytorch side of huggingface transformers library (rename?) -`huggingface_tf` is really the Keras side of huggingface transformers library (rename?) -`torch_hub_vision` is vision models from Torch Hub, but we also have non-vision models from Torch Hub in `torch_hub` (converge?) -`stable_diffusion` is some stable diffusion sub-models from huggingface diffusers library (rename to `diffusers`? we should probably get more models from `diffusers` as well!)
> > edit: I also want to bring in the "popular on huggingface" models - where do those fit into this dir structure?
> 
> @jeremyfowers I addressed all of the suggestions regarding renaming/moving files.
> 
> What do you think we should do about adding the popular models? A more updated list that what we have would be nice. How many models should we add here? Also, is that within the scope of this MR?

Not in the scope of this MR, filed #59. Let's discuss offline :)

**Review by jeremyfowers — APPROVED** [review]

File: models/graph_convolutions/appnp.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# labels: test_group::mlagility name::appnp author::graph_convolutions
```
Comment (jeremyfowers) [inline models/graph_convolutions/appnp.py:1]: Move all of these to `graph_convolutions/skip/*`

File: src/mlagility/parser.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import argparse
```
Comment (jeremyfowers) [inline src/mlagility/parser.py:1]: add a requirements.txt to `models/.`

Comment (danielholanda) [reply src/mlagility/parser.py:1]: Added!

File: models/graph_convolutions/appnp.py (hunk 2 above)
Comment (danielholanda) [reply models/graph_convolutions/appnp.py]: moved!

**Pull request merged**